tests/gnrc_dhcpv6_client_stateless/tests-as-root/01-run.py

```diff
@@ -0,0 +1,173 @@
+        mtu = ICMPv6NDOptMTU()
+        prefix_info = ICMPv6NDOptPrefixInfo(prefix=self.prefix, prefixlen=64)
+        sendp(
+            header / ra / src_ll_addr / mtu / prefix_info,
+            iface=self.iface,
+            verbose=False,
+        )
+
+    @ATMT.state()
+    def WAITING_FOR_DHCP_IR(self):
+        pass
+
+    @ATMT.receive_condition(WAITING_FOR_DHCP_IR, prio=1)
+    def received_DHCPv6(self, pkt):
+        if DHCP6_InfoRequest in pkt:
+            hwaddrs = get_hwaddrs(self.child)
+
+            assert DHCP6OptClientId in pkt and DUID_LL in pkt[DHCP6OptClientId].duid
+            assert pkt[DHCP6OptClientId].duid[DUID_LL].lladdr in hwaddrs
+
+            # The information-request contained a MUD URL option
+            assert DHCP6OptMudUrl in pkt
+            mud_option = pkt[DHCP6OptMudUrl]
+            assert mud_option.optlen == len(MUD_TEST_URL)
+
+            if mud_option_loaded:
+                assert mud_option.mudstring == MUD_TEST_URL
+            else:
+                assert mud_option.optcode == MUD_OPTION_CODE
+                assert mud_option.data == MUD_TEST_URL
+
+            raise self.END().action_parameters(pkt)
+
+    @ATMT.action(received_DHCPv6)
+    def on_DHCPv6_IR(self, pkt):
+        self.send_DHCPv6_Reply(pkt)
+
+    def send_DHCPv6_Reply(self, pkt):
+        header = build_reply_headers(pkt)
+        trid = pkt[DHCP6_InfoRequest].trid
+        srv_duid = header[Ether].src
+        cli_id = DHCP6OptClientId(duid=pkt[DHCP6OptClientId].duid)
+        srv_id = DHCP6OptServerId(duid=DUID_LL(lladdr=srv_duid))
+        sendp(
+            header / DHCP6_Reply(trid=trid) / cli_id / srv_id,
+            iface=self.iface,
+            verbose=False,
+        )
+
+    @ATMT.timeout(WAITING_FOR_NDP_RS, 10.0)
+    @ATMT.timeout(WAITING_FOR_DHCP_IR, 10.0)
+    def waiting_timeout(self):
+        raise self.ERROR_TIMEOUT()
+
+    @ATMT.state(final=1)
+    def END(self):
+        time.sleep(1)
+
+        # check if global address was configured
+        self.child.sendline("ifconfig")
+        # remove one trailing ':' from prefix just to be safe ;-)
+        self.child.expect(r"inet6 addr:\s+{}[0-9a-fA-F:]+\s".format(self.prefix[:-1]))
+        print("SUCCESS")
+
+
+def get_hwaddrs(child):
+    hwaddrs = []
+    child.sendline("ifconfig")
+    child.expect(r"HWaddr:\s+(([A-Fa-f0-9]{2}:?)+)\s")
+    hwaddrs.append(child.match.group(1).lower())
+    if len(hwaddrs[0]) == 5:  # short address
+        res = child.expect([pexpect.TIMEOUT, r"Long HWaddr:\s+(([A-Fa-f0-9]{2}:?)+)\s"])
+        if res > 0:
+            hwaddrs.append(child.match.group(1).lower())
+    return hwaddrs
+
+
+def build_router_advertise_header(pkt):
+    return Ether() / IPv6(dst="ff02::1")
+
+
+def build_reply_headers(pkt):
+    src_ether = pkt[Ether].src
+    src_ip = pkt[IPv6].src
+    sport = pkt[UDP].sport
+    dport = pkt[UDP].dport
+    return Ether(dst=src_ether) / IPv6(dst=src_ip) / UDP(sport=dport, dport=sport)
```

Comment:
Since these are related to the `send_` functions, I think they can be part of the Automaton class (as `@staticmethod`s).

Reply:
I converted them to static methods now :)